File: 1486.数组异或操作.py
#
# @lc app=leetcode.cn id=1486 lang=python3
#
# [1486] 数组异或操作
#

# @lc code=start
class Solution:
    def xorOperation(self, n: int, start: int) -> int:
        xor = 0
        for i in range(n):
            xor ^= (start + i * 2 )
        return xor

# @lc code=end

File: 1486.数组异或操作.cpp
/*
 * @lc app=leetcode.cn id=1486 lang=cpp
 *
 * [1486] 数组异或操作
 */

// @lc code=start
class Solution {
public:
    int xorOperation(int n, int start) {
        int count = 0;
        int sum = 0;
        while(count < n)
        {
            sum ^= start + 2*count;
            count++;
        }
        return sum;
    }
};
// @lc code=end

